### DISCUSSION TITLE
Problema con continui riavii container HA e Z2M su RASP 4B

### DISCUSSION THREAD

**Discussion by braudoo** [indomus/forum#2706] [GENERALE - Discussioni generiche]
Ciao a tutti, da un pò di tempo i container di HA e Z2M tendono ad uscire e a riavviarsi (secondo la policy impostata).
Purtroppo non riesco a spiegarmi questo comportamento, e guardando i file di log non riesco a capire cosa porti a queste uscite/riavvi.
Ho installato i container su Rasp 4B. Entrambi container aggiornati all'ultima versione.

Partendo da HA, questo l'estratto del log prima del riavvio:

```
2023-11-09 08:13:36.030 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration scheduler which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-11-09 08:13:54.207 WARNING (MainThread) [homeassistant.config] Package alexa_google_text_1_2 contains invalid customize
2023-11-09 08:13:54.208 WARNING (MainThread) [homeassistant.config] Package lavatrice2 contains invalid customize
2023-11-09 08:14:02.612 WARNING (Recorder) [homeassistant.components.recorder.util] The system could not validate that the sqlite3 database at //config/home-assistant_v2.db was shutdown cleanly
2023-11-09 08:14:03.227 WARNING (Recorder) [homeassistant.components.recorder.util] Ended unfinished session (id=140 from 2023-11-06 22:23:44.810154)
```

Mentre per Z2M, ho notato che mi da un exit code 1, dove in diversi forum attribuiscono il motivo o al dongle usb o alla versione dell'immagine Z2M:
```
Zigbee2MQTT:error 2023-11-11 07:52:10: Exception while calling fromZigbee converter: 'illuminance_lux_calibration' is not a number, got string ()}
Zigbee2MQTT:info  2023-11-11 07:53:37: MQTT publish: topic 'zigbee2mqtt/sensore_human_body', payload '{"battery":100,"device":{"applicationVersion":5,"dateCode":"20170627","friendlyName":"sensore_human_body","ieeeAddr":"0x00158d0008369b45","manufacturerID":4151,"manufacturerName":"LUMI","model":"RTCGQ11LM","networkAddress":11281,"powerSource":"Battery","softwareBuildID":"3000-0001","stackVersion":2,"type":"EndDevice","zclVersion":1},"device_temperature":26,"illuminance":0,"illuminance_lux":0,"linkquality":255,"occupancy":false,"power_outage_count":28,"voltage":3025}'
Zigbee2MQTT:info  2023-11-11 07:54:24: MQTT publish: topic 'zigbee2mqtt/luce_balcone', payload '{"brightness":254,"device":{"applicationVersion":2,"dateCode":"20201012","friendlyName":"luce_balcone","hardwareVersion":1,"ieeeAddr":"0x001788010b2735b4","manufacturerID":4107,"manufacturerName":"Signify Netherlands B.V.","model":"929001821618","networkAddress":31151,"powerSource":"Mains (single phase)","softwareBuildID":"1.76.10","stackVersion":1,"type":"Router","zclVersion":2},"linkquality":255,"power_on_behavior":null,"state":"OFF","update":{"installed_version":16785162,"latest_version":16785162,"state":"idle"},"update_available":false}'
Zigbee2MQTT:info  2023-11-11 07:56:40: MQTT publish: topic 'zigbee2mqtt/smart_switch', payload '{"action":null,"battery":100,"click":null,"device":{"applicationVersion":2,"dateCode":"20180525","friendlyName":"smart_switch","hardwareVersion":2,"ieeeAddr":"0x00158d0007e06913","manufacturerID":4151,"manufacturerName":"LUMI","model":"WXKG11LM","networkAddress":49824,"powerSource":"Battery","softwareBuildID":"3000-0001","stackVersion":2,"type":"EndDevice","zclVersion":1},"device_temperature":27,"linkquality":255,"power_outage_count":52,"voltage":3035}'
Zigbee2MQTT:error 2023-11-11 08:11:32: Not connected to MQTT server!
Error: EIO: i/o error, write
```

Chiedo a chi piu esperto di me su docker e container quali possano essere le cause di questi riavvi.

![Screenshot_2023-11-11-17-18-52-53_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/indomus/forum/assets/58866563/1fa1bb0a-be4b-4a79-9532-4455ca8edaa3)

**Comment by snakuzzo** [comment]
per HA non mi pare ci sia nulla di così parlante. Quel warning sul db potrebbe indicare qualche problema a livello di srittura, ma non è detto.

L'exit code 1 di zigbee2mqtt non è legato al dongle, o comunque non è un errore così specifico. Si tratta di un errore che indica il fallimento dell'applicazione. Tutti gli exit code con codice diverso da 0 indicano solitamente che l'applicazione è terminata con errore.
Se guardi l'ultima riga del log, risulta un errore `i/o error, write` e, poco prima, una disconnessione dal broker mqtt (probabilmente fallito pure lui?).

Considerando che HA non crasha mai, l'unica cosa che mi verrebbe da pensare è qualche problema sul supporto di scrittura. Non so se sei la stessa persona che qualche giorno fa su Telegram ha detto di avere questi problemi, ma in tal caso avevo già detto di controllare i log su `/var/log/syslog` o comunque i log sotto `/var/log`

P.S. il markdown come l'hai usato ora (con un backtick) si usa quando devi postare qualcosa su una sola riga. Dovevi usare il markdown con i tre backtick. Correggi!

**Comment by braudoo** [comment]
Ciao dopo poco la sd è crashata, quindi il problema era sul supporto di scrittura.
A breve migreró su SSD.
Grazie per il supporto.

**Reply by snakuzzo** [reply]
Come Volevasi Dimostrare
